transport/http/agent_test.go

```diff
@@ -4,374 +4,256 @@ import (
-			s.msg = "no notification after release"
-			s.expectedPeers = []peerExpectation{}
-
-			sub := transporttest.NewMockPeerSubscriber(mockCtrl)
-			sub.EXPECT().NotifyStatusChanged(gomock.Any()).Times(0)
-
-			pid := hostport.PeerIdentifier("localhost:1234")
+		{
+			msg:         "one retain",
+			identifiers: []string{"i1"},
+			subscriberDefs: []SubscriberDefinition{
+				{ID: "s1"},
+			},
+			actions: []AgentAction{
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s1", ExpectedPeerID: "i1"},
+			},
+			expectedPeers: []peerExpectation{
+				{id: "i1", subscribers: []string{"s1"}},
+			},
+		},
+		{
+			msg:         "one retain one release",
+			identifiers: []string{"i1"},
+			subscriberDefs: []SubscriberDefinition{
+				{ID: "s1"},
+			},
+			actions: []AgentAction{
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s1", ExpectedPeerID: "i1"},
+				ReleaseAction{InputIdentifierID: "i1", InputSubscriberID: "s1"},
+			},
+		},
+		{
+			msg:         "three retains",
+			identifiers: []string{"i1"},
+			subscriberDefs: []SubscriberDefinition{
+				{ID: "s1"},
+				{ID: "s2"},
+				{ID: "s3"},
+			},
+			actions: []AgentAction{
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s1", ExpectedPeerID: "i1"},
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s2", ExpectedPeerID: "i1"},
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s3", ExpectedPeerID: "i1"},
+			},
+			expectedPeers: []peerExpectation{
+				{id: "i1", subscribers: []string{"s1", "s2", "s3"}},
+			},
+		},
+		{
+			msg:         "three retains one release",
+			identifiers: []string{"i1"},
+			subscriberDefs: []SubscriberDefinition{
+				{ID: "s1"},
+				{ID: "s2r"},
+				{ID: "s3"},
+			},
+			actions: []AgentAction{
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s1", ExpectedPeerID: "i1"},
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s2r", ExpectedPeerID: "i1"},
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s3", ExpectedPeerID: "i1"},
+				ReleaseAction{InputIdentifierID: "i1", InputSubscriberID: "s2r"},
+			},
+			expectedPeers: []peerExpectation{
+				{id: "i1", subscribers: []string{"s1", "s3"}},
+			},
+		},
+		{
+			msg:         "three retains, three release",
+			identifiers: []string{"i1"},
+			subscriberDefs: []SubscriberDefinition{
+				{ID: "s1"},
+				{ID: "s2"},
+				{ID: "s3"},
+			},
+			actions: []AgentAction{
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s1", ExpectedPeerID: "i1"},
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s2", ExpectedPeerID: "i1"},
+				RetainAction{InputIdentifierID: "i1", InputSubscriberID: "s3", ExpectedPeerID: "i1"},
+				ReleaseAction{InputIdentifierID: "i1", InputSubscriberID: "s1"},
+				ReleaseAction{InputIdentifierID: "i1", InputSubscriberID: "s2"},
+				ReleaseAction{InputIdentifierID: "i1", InputSubscriberID: "s3"},
+			},
+		},
+		{
+			msg:         "no retains one release",
```

Comment:
These messages seem to only describe how the world was setup, they don't say anything about the expected behavior. I would expect something more like:

```
err no reference when release called on empty peer list
```

The same seems to go for alot of these messages.